akka-stream-tests/src/test/scala/akka/stream/FusingSpec.scala

```diff
@@ -84,6 +84,45 @@ class FusingSpec extends StreamSpec {
       refs.toSet should have size (in.size + 1) // outer/main actor + 1 actor per subflow
     }
 
+    "propagate downstream errors through async boundary" in {
+      val slowInitSrc = Source
+      //todo: can this be replaced with some promise+future composition?
+        .unfoldResource(() => { Thread.sleep(800) }, (_: Unit) => Some(1), (_: Unit) => ())
```

Comment:
How about:

```suggestion
      import system.dispatcher
      val slowInitSrc =
        Source
          .futureSource(akka.pattern.after(800.millis, system.scheduler)(Future.successful(Source.repeat(1))))
```

also below.

Reply:
Could it be done using a latch instead of timing based? Opening it after the first assert below (if I read this right)?

Reply:
Yeap, I understand that it must block, a latch would block as well. The reason I asked is if it could be a way to make the test faster and less sensitive to CI server instability.

Reply:
ok, seems there's a small detail I've missed here. this graph actually materializes into three actors since `Source.unfoldResources` uses the io dispatcher by default.
So when I've tried @johanandren's proposal of using a latch it turned out that the first watch termination (the `f1` future) was already successfully completed before the latch was released, actually the fact it successfully completed confirms the bug.
I'll try to use another source and c if I can get this reproduced in a reliable manner.